### PR TITLE
Use Docker composer to run all components into its own container

### PR DESCRIPTION
**Pull request by iprunache**
## Why
Docker will allow us to easily distribute ready to run containers for every CAMI component.

## What
- [ ] create Docker container for the following components:
   - [x] MySQL instance
   - [x] REST API
   - [x] RabbitMQ instance
   - [ ] service daemon
- [x] use Docker composer to get all containers to work together.

### CODE REVIEW

**Comment by iprunache** [comment]
@aismail Please give this a look when you have time! I could really use the feedback since from my point of view the current setup needs considerable improvement :)

File: README.md [hunk 1]
```diff
@@ -1,7 +1,79 @@
...
+
+The project now has a Docker containers configured for the microservices. This is still WIP but a functional dev env can be started using Docker.
+
+You will need `docker-compose` to get the env running. Follow the [installation guide](https://docs.docker.com/compose/install/).
+
+Once you have `docker-compose` installed just run this from the project root:
+```
+docker-compose up
+```
+This will start all cami microservices and output their standard output. It may take a while for all the containers to come online especially on the first run when the mysql database is initialised on cami-store. To check that all is working try this in your browser: `http://127.0.0.1:8000/api/v1/medication-plans/`. Replace `127.0.0.1` with your VM's ip if you're running Docker in a VM.
```
Comment (aismail) [inline README.md]: @iprunache why would anyone run Docker in a VM nowadays? We have a native Docker implementation for each major OS (Linux, MacOS, Windows).

File: docker/cami-medical-compliance/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+FROM williamyeh/ansible:ubuntu16.04
```
Comment (aismail) [inline docker/cami-medical-compliance/Dockerfile]: @iprunache I'm a bit surprised that we're not using the 'official' ubuntu images (https://hub.docker.com/_/ubuntu/), so I'm guessing this must be something related to Ansible support. If that's the case, a short comment would be great!

File: README.md (hunk 1 above)
Comment (iprunache) [inline README.md]: Native support seems to be new. Maybe someone does not have OS X 10.10 :)

File: docker/cami-medical-compliance/docker-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+# Using hardcoded values for paths for now.
+# TODO(@iprunache): find a way to make the paths configurable
+
+# Doing a migration at the container start is not a good idea on the long term
+# as usually one wants containers to start as fast as possible.
+python /cami-project/medical_compliance/manage.py migrate
```
Comment (aismail) [inline docker/cami-medical-compliance/docker-entrypoint.sh]: @iprunache that's one reason, and another reason is the principle of least surprise - you wouldn't expect a server starting up to run migrations. In production, you usually want to control very tightly when migrations are run, and it's usually in maintenance mode for the whole product if there's a major schema change.

File: docker/cami-rabbitmq/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
+FROM rabbitmq:3-management
+
+# Bake configuration into image for ease of use. We'll later move this to
```
Comment (aismail) [inline docker/cami-rabbitmq/Dockerfile]: 👍

File: docker/cami-store/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,16 @@
+FROM mysql:5.6
+
+# Add the store schema to the container to be run when the first container is
+# run. This requires that the build context is the project root folder. Run
+# build like:
+# `docker build -f docker/cami-store/Dockerfile .`
+ADD store/cami.sql /docker-entrypoint-initdb.d/
+
+# Hardcode passwors until we learn a better way to do this.
```
Comment (aismail) [inline docker/cami-store/Dockerfile]: I think people will either use a service to distribute secrets (not sure how it's done in Docker right now), or hardcode the ENV vars in docker-compose.yml (this is what I've seen in some of our clients' stacks). We definitely need to dig up a better solution, I don't think hardcoding in docker-compose.yml is that good either.

File: docker/cami-medical-compliance/Dockerfile (hunk 2 above)
Comment (iprunache) [inline docker/cami-medical-compliance/Dockerfile]: Yes, it's a base image with Ansible installed. Ansible do not maintain such images themselves anymore so we're using what's most popular.

File: docker/cami-medical-compliance/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+# or, for example, FROM ansible/ubuntu14.04-ansible:stable
+
+# Add playbooks to the Docker image. This requires that the build context is the
+# project root folder. Run build like:
+# `docker build -f docker/cami-store/Dockerfile .`
+COPY . /cami-project
```
Comment (aismail) [inline docker/cami-medical-compliance/Dockerfile]: @iprunache I don't think this is a standard filesystem layout. Take a look here: https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard. It's quite an interesting read, IMO.

Comment (iprunache) [inline docker/cami-medical-compliance/Dockerfile]: It's a reminiscence from how Vagrant did it with it's base folder. A better place might would be /opt/cami-project.

File: docker/cami-store/Dockerfile (hunk 5 above)
Comment (iprunache) [inline docker/cami-store/Dockerfile]: I read something about such a service: `confd`. It seems to allow creating template config files on the container which it populates with ENV vars taken from the container run command.

**Pull request merged**